src/Analyzers/Core/CodeFixes/NamingStyle/NamingStyleFixAllProvider.cs

```diff
@@ -0,0 +1,121 @@
+using Microsoft.CodeAnalysis.NamingStyles;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CodeFixes.NamingStyles
+{
+    internal partial class NamingStyleCodeFixProvider
+    {
+        private class NamingStyleCodeFixAllProvider : FixAllProvider
+        {
+            public static readonly NamingStyleCodeFixAllProvider Instance = new();
+
+            private NamingStyleCodeFixAllProvider() { }
+
+            public override async Task<CodeAction?> GetFixAsync(FixAllContext fixAllContext)
+            {
+                var diagnostics = fixAllContext.Scope switch
+                {
+                    FixAllScope.Document when fixAllContext.Document is not null => await fixAllContext.GetDocumentDiagnosticsAsync(fixAllContext.Document).ConfigureAwait(false),
+                    FixAllScope.Project => await fixAllContext.GetAllDiagnosticsAsync(fixAllContext.Project).ConfigureAwait(false),
+                    FixAllScope.Solution => await GetSolutionDiagnosticsAsync(fixAllContext).ConfigureAwait(false),
+                    _ => default
+                };
+
+                var keys = fixAllContext.CodeActionEquivalenceKey!.Split('|');
+                RoslynDebug.Assert(keys.Length == 2);
+                RoslynDebug.Assert(Guid.TryParse(keys[0], out _));
+                RoslynDebug.Assert(int.TryParse(keys[1], out _));
+
+                var symbolSpecificationId = keys[0];
+                var diagnosticsToFix = diagnostics.WhereAsArray(
+                    d => d.Location.IsInSource && d.Properties["SymbolSpecificationID"] == symbolSpecificationId);
+                if (diagnosticsToFix.IsDefaultOrEmpty)
+                {
+                    return null;
+                }
+
+                var complaintNameIndex = int.Parse(keys[1]);
+                var symbolsToRename = await GetSymbolsToRenameAsync(
+                    fixAllContext.Solution, diagnosticsToFix, complaintNameIndex, fixAllContext.CancellationToken).ConfigureAwait(false);
+                if (symbolsToRename.IsEmpty)
+                {
+                    return null;
+                }
+
+                return new CustomCodeActions.SolutionChangeAction(
+                    CodeFixesResources.Fix_all_name_violations,
+                    c => FixAllAsync(fixAllContext.Solution, symbolsToRename, c),
+                    fixAllContext.CodeActionEquivalenceKey);
+            }
+
+            private static Task<Solution> FixAllAsync(Solution solution, ImmutableHashSet<(ISymbol symbol, string newName)> _1, CancellationToken _2)
+            {
+                // TODO:
+                // Renamer needs a new API which could rename multiple symbols using the same solution snapshot
+                // and calculate the changed documents for each symbol so that we could notify IRefactorNotifyService
+                return Task.FromResult(solution);
```

Comment:
I assume that this PR should be marked as "in progress"? until this rename API is in we can't merge thsi PR right?